File: tools/listen.py
import socket
import binascii
import struct

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind(('', 50000))
s.listen(1)

connection, address = s.accept()

print 'Connection from', address

while 1:
	data = connection.recv(1024)
	if not data: break
	while len(data) > 0:
		#print '%s' % struct.unpack('>H', data)
		datum = struct.unpack('!H', data[0:2])
		pan = (datum[0] & 0x7800) >> 9;
		
		print 'Value: %H' % datum[0]
		print 'Pan: %H' % pan
		
		data = data[2:len(data)-2]
connection.close()

File: tools/listen.php
<?php

$socket = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
echo "Create: " . socket_strerror(socket_last_error($socket));

socket_set_block($socket);
echo "Set block: " . socket_strerror(socket_last_error($socket));

socket_bind($socket, '127.0.0.1', 50000);
echo "Bind: " . socket_strerror(socket_last_error($socket));

socket_listen($socket);
echo "Listen: " . socket_strerror(socket_last_error($socket));

$data = '';
$r = array($socket);
$w = null;
$e = null;

socket_select($r, $w, $e, 0);
socket_accept($socket);

do
{
	$r = array($socket);
	$w = null;
	$e = null;
	socket_select($r, $w, $e, 0);
	$length = @socket_recv($socket, $data, 1024, 0);
	printf('%s', $data);
} while ($length !== 0);

socket_close($socket);
